recipe-server/normandy/base/middleware.py

```diff
@@ -91,3 +99,32 @@ class NormandyCommonMiddleware(CommonMiddleware):
     """
 
     response_redirect_class = HttpResponsePermanentRedirectCached
+
+
+class NormandySecurityMiddleware(SecurityMiddleware):
+    """Logs HTTP to HTTPS redirects, and adds cache headers to them."""
+
+    def process_request(self, request):
+        response = super().process_request(request)
+        if response is not None:
+            assert type(response) is http.HttpResponsePermanentRedirect
+            patch_cache_control(response, public=True, max_age=settings.PERMANENT_REDIRECT_CACHE_TIME)
+
+            # Pull out just the HTTP headers from the rest of the request meta
+            headers = {
+                key.lstrip('HTTP_'): value
+                for (key, value) in request.META.items()
+                if key.startswith('HTTP_') or key == 'CONTENT_TYPE' or key == 'CONTENT_LENGTH'
+            }
+
+            logger.debug(
```

Comment:
Could this collect any personally-identifiable or otherwise sensitive info? CSRF tokens? Auth0 information? I think generally we should avoid storing that kind of data in the logs when possible (although we may already be storing that info in the logs and I'm just not aware or wrong that it's a potential issue).

Perhaps we should consider a whitelist of headers to collect, and no request body data.

Reply:
You're right, this can collect personally-identifiable information. This is here specifically because @relud asked for "absolutely everything you can get, especially body and headers".

Also, our logs are already considered sensitive information. Some information gets scrubbed out before it gets put in something like parquet.

@relud does that sound correct?

Reply:
yes. also in prod you aren't supposed to be able to hit this page.